File: rust/src/371.rs
//371. Sum of Two Integers
impl Solution {
    pub fn get_sum(a: i32, b: i32) -> i32 {
        let mut x = a ^ b; // xor for sum
        let mut y = (a & b) << 1; // & for carry

        while y != 0 {
            let car = (x & y) << 1;
            x ^= y;
            y = car;
        }

        x
    }
}
//sub: https://leetcode.com/problems/sum-of-two-integers/submissions/1577500011/
//sol: https://leetcode.com/problems/sum-of-two-integers/solutions/6549813/371-sum-of-two-integers-by-nalindalal200-5brn/

File: ts/src/371.ts
//3541. Find Most Frequent Vowel and Consonant
function maxFreqSum(s: string): number {
  const freq: Record<string, number> = {};

  for (const ch of s.toLowerCase()) {
    if (/[a-z]/.test(ch)) {
      freq[ch] = (freq[ch] || 0) + 1;
    }
  }

  const isVowel = (c: string) => "aeiou".includes(c);

  let maxVowel = 0;
  let maxConsonant = 0;

  for (const [ch, count] of Object.entries(freq)) {
    if (isVowel(ch)) {
      maxVowel = Math.max(maxVowel, count);
    } else {
      maxConsonant = Math.max(maxConsonant, count);
    }
  }

  return maxVowel + maxConsonant;
}
//sub: https://leetcode.com/problems/find-most-frequent-vowel-and-consonant/submissions/1768790793/?envType=daily-question&envId=2025-09-13
//sol: https://leetcode.com/problems/find-most-frequent-vowel-and-consonant/solutions/7183971/3541-find-most-frequent-vowel-and-conson-9rwe/
